components/vc/front/src/torque.c

```diff
@@ -47,13 +54,167 @@ static struct
+    const bool torque_inc_active = (CANRX_get_signal(VEH, SWS_requestTorqueInc, &torque_change_request) != CANRX_MESSAGE_SNA) &&
+                                   (torque_change_request == CAN_DIGITALSTATUS_ON);
+    const bool torque_dec_active = (CANRX_get_signal(VEH, SWS_requestTorqueDec, &torque_change_request) != CANRX_MESSAGE_SNA) &&
+                                   (torque_change_request == CAN_DIGITALSTATUS_ON);
+
+    if (torque_inc_active ^ torque_dec_active)
+    {
+        const drv_timer_state_E timer_state = drv_timer_getState(&torque_data.torque_change_timer);
+        if (timer_state == DRV_TIMER_STOPPED)
+        {
+            drv_timer_start(&torque_data.torque_change_timer, TORQUE_CHANGE_DELAY);
+            torque_request_max = torque_inc_active ? torque_request_max + 1 : torque_request_max - 1;
+            torque_request_max = SATURATE(MIN_TORQUE_RANGE, torque_request_max, ABSOLUTE_MAX_TORQUE);
+        }
+        else if (timer_state == DRV_TIMER_EXPIRED)
+        {
+            drv_timer_stop(&torque_data.torque_change_timer);
+        }
+
+        torque_data.torque_request_max = torque_request_max;
+    }
+    else
+    {
+        drv_timer_stop(&torque_data.torque_change_timer);
+    }
+
+    return torque_request_max;
+}
+
+static void evaluate_launch_control(float32_t accelerator_position, float32_t brake_position)
+{
+#if FEATURE_IS_ENABLED(FEATURE_LAUNCH_CONTROL)
+    CAN_digitalStatus_E launch_control_requested = CAN_DIGITALSTATUS_SNA;
+    const bool launch_control_request_was_active = torque_data.torque_control_request_active;
+    torque_data.torque_control_request_active = (CANRX_get_signal(VEH, SWS_requestLaunchControl, &launch_control_requested) != CANRX_MESSAGE_SNA) &&
+                                                (launch_control_requested == CAN_DIGITALSTATUS_ON);
+    const bool launch_control_request_rising = !launch_control_request_was_active && torque_data.torque_control_request_active;
+
+    switch (torque_data.launch_control_state)
+    {
+        case LAUNCHCONTROL_REJECTED:
+            if (drv_timer_getState(&torque_data.launch_control_timer) == DRV_TIMER_EXPIRED)
+            {
+                torque_data.launch_control_state = LAUNCHCONTROL_INACTIVE;
+            }
+            __attribute__((fallthrough));
+        case LAUNCHCONTROL_INACTIVE:
+            if (launch_control_request_rising)
+            {
+                if ((brake_position > LAUNCH_CONTROL_BRAKE_THRESHOLD) &&
+                    (accelerator_position < LAUNCH_CONTROL_THROTTLE_START_CUTOFF))
+                {
+                    torque_data.launch_control_state = LAUNCHCONTROL_HOLDING;
+                    drv_timer_stop(&torque_data.launch_control_timer);
+                }
+                else
+                {
+                    torque_data.launch_control_state = LAUNCHCONTROL_REJECTED;
+                    drv_timer_start(&torque_data.launch_control_timer, LAUNCH_CONTROL_REJECTED_MS);
```

Comment:
maybe rename this timer? `launch_control_lockout_timer` maybe? Also I like to put the expected period in the timer name so you don't accidentally poll it from a function running at the wrong period (which would mess up the timer).

Also, aren't we camel case in this context? Shouldn't this be something like `launchControlLockoutTimer_10Hz`